polytracker/taint_dag.py

```diff
@@ -174,7 +175,7 @@ def decode_node(self, label: int) -> TDNode:
                 return TDRangeNode(v1, v2, affects_cf)
 
     @property
-    def nodes(self) -> Generator[TDNode, None, None]:
+    def nodes(self) -> Iterator[TDNode]:
         assert self.header.tdag_mapping_offset > 0
         assert self.header.tdag_mapping_size > 0
```

Comment:
Suggest that we move the assertions to the __init__-function to get a early error if there is something wrong with the tdag-file.